sklearn/manifold/_spectral_embedding.py

```diff
@@ -315,8 +315,6 @@ def spectral_embedding(
         # problem.
         if not sparse.issparse(laplacian):
             warnings.warn("AMG works better for sparse matrices")
-        # lobpcg needs double precision floats
-        laplacian = check_array(laplacian, dtype=np.float64, accept_sparse=True)
```

Comment:
we still want to check that the inputs are valid on public API.

```python
laplacian = check_array(laplacian, dtype=[np.float64, np.float32], accept_sparse=True)
```

Reply:
make sense, indeed, thanks!